cluster-autoscaler/context/autoscaling_context.go

```diff
@@ -51,6 +51,16 @@ type AutoscalingContext struct {
 	LogRecorder *utils.LogEventRecorder
 }
 
+// GpuLimits define lower and upper bound on GPU instances of given type in cluster
+type GpuLimits struct {
```

Comment:
This is tied to my comment in main.go. Obviously structured information is more useful than a bunch of strings, but I'm not convinced it should be the responsibility of the code creating AutoscalingOptions, or code using them to actually parse this stuff. Of course right now we use both approaches, which is the worst option :( (parse before creating AutoscalingOptions for limits, parse when creating stuff using AutoscalingOptions for other stuff, for example autodiscovery config).

Reply:
IDK. It seems good direction to me. And it is obvious that if we want to move into this direction there will be a phase (probably never ending) where some options are still passed not in a way we would like.
I would not call it worst option. We cannot fix everything. Is that a reason not to fix some things (those we are working on at the moment). 

Above assume that we agree that it is actually direction we want to move on :)